### PR TITLE
make packet commitments fixed length

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

closes: #7563 

It makes commit bytes with a fixed length (32 bytes) for better gas cost in Solidity. I also added some tests to document the output, as we are using those outputs for tests in the Solidity implementation as well to make sure they match.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by chatton — APPROVED** [review]
LGTM!

File: modules/core/04-channel/v2/types/commitment.go [hunk 1]
```diff
@@ -27,8 +27,10 @@ func CommitPacket(packet Packet) []byte {
 	appHash := sha256.Sum256(appBytes)
 	buf = append(buf, appHash[:]...)
 
+	buf = append([]byte{byte(2)}, buf...)
+
 	hash := sha256.Sum256(buf)
-	return append([]byte{byte(2)}, hash[:]...)
+	return hash[:]
```
Comment (DimitrisJim) [inline modules/core/04-channel/v2/types/commitment.go:33]: can we amend documentation to reflect the fact that the hash includes the prefix?

**Review by DimitrisJim — APPROVED** [review]
nice! :sparkles:

**Review by AdityaSripal — APPROVED** [review]
Ack, thanks for the tests as well!

Modulo doc changes suggesyed by @DimitrisJim

**Pull request merged**